### PR TITLE
[MINI-5634] Back to display by url when exit external webview

### PR DESCRIPTION
**Pull request by shohiebsense**
# Description
on the external webview page caused by a redirect url, when back navigation is pressed, should go back to 'display by url' page.
Enables MiniApp url scheme.

## Links
[MINI-5634](https://jira.rakuten-it.com/jira/browse/MINI-5634)

## Video
https://rak.box.com/s/p045g5sevymqr75mm7vm2b111s00bsl1
https://rak.box.com/s/6jzkw1bedxqta0rt51nt63dpnn8waka5 (obsolete)

MiniApp by deeplink 
https://rak.box.com/s/6jzkw1bedxqta0rt51nt63dpnn8waka5

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (munir-rakuten) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:462]: is this the correct behaviour? we are opening an external url from miniapp to a external webview, closing the external webview should not close the miniapp it self. it should just close the external webview and come back to the miniapp screen. that's why there was not any `else` statement. please revert this to the original one.

**Review by munir-rakuten — COMMENTED** [review]
Please check my comment.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: from this screen: https://rak.box.com/s/8su7ea88scov85798ztduv0moge800me
assuming you have checked the [issue video,](https://rak.app.box.com/s/n98c8n6baqpiucotfn9bpzgloydfa33w), if the issue is still a bug, then instead I made some arguments to make sure that it's from MiniAppByUrlActivity. Please check.

Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: @shohiebsense i have checked the video, looks like it behaved correctly, the url qa is opening is somehow not correct or a `Link` deeplink, that's why it opened the miniapp first(which failed to load)then miniapp opens the google play store in a external webview. So closing the external webview should come back to the miniapp which is failed to load so showing a empty screen, then back should close the miniapp screen.

Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: @munir-rakuten the url is provided by QA should be correct. I verify myself, this is it https://rak.box.com/s/0ai07421oibuykjl6ca905uiopo5tsq8 if the activity is not finished.

Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: @shohiebsense the url must be a miniapp url to load, not a `google.com`, if the url is not a miniapp url it will open the external webview, which is happening in your video, if external webview back close the miniapp also then when we will open a link from miniapp it will be affected as well.

Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: This url https://one.rakuten.co.jp/miniapp/preview/404e46b4-263d-4768-b2ec-8a423224bead is not correct url for mini app? @munir-rakuten

Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:472]: `one.rakuten.co.jp` is not a valid url for demo app. it's a deeplink is for link app and it's a deeplink, `mini app by url` should not open deeplinks.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:469]: Usually, "isClosedByBackPressed" is triggered from `WebViewActivity` with `true` which means we can close the respected screens if there is nothing to emit result. So, I think, we can do following:

```suggestion
                } else finish()
```

Then, we don't need any other changes e.g. `isFromMiniAppByUrlActivityTag`

Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:469]: Nice one then, thanks for confirming. I can go back to use this one once get at least one approval

File: testapp/src/main/AndroidManifest.xml
Comment (shohiebsense) [reply testapp/src/main/AndroidManifest.xml:88]: fixed.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:135]: Is it possible to pass this parameter through `startUrl(...)`, so it would be expressible to outside of this class e.g.

```
        fun startUrl(
            context: Context,
            appUrl: String,
            miniAppSdkConfig: MiniAppSdkConfig? = null,
            updatetype: Boolean = false,
            isFromMiniAppByUrlActivity: Boolean = true
        ) { ... }
```

**Review by khairul-alam-licon — COMMENTED** [review]
Looking good, I have left few suggestions.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/helper/AppHelper.kt
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/helper/AppHelper.kt]: it's reverted now. would not add this

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt]: thanks for the catch

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/input/MiniAppByUrlActivity.kt
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/input/MiniAppByUrlActivity.kt]: aa oopsie, already reverted back

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt]: I thought about it this way, it just I don't think it could scale to the point that it needs to be this way, ok changed.

**Review by khairul-alam-licon — APPROVED** [review]
Looks good!
Thank you for your work 👏

**Review by munir-rakuten — APPROVED** [review]
LGTM, thanks for your effort👍

**Pull request merged**
